unix/w0vncserver/wlr/WlrVirtualKeyboard.cxx

```diff
@@ -58,9 +68,34 @@ WlrVirtualKeyboard::~WlrVirtualKeyboard()
     zwp_virtual_keyboard_manager_v1_destroy(manager);
   if (keyboard)
     zwp_virtual_keyboard_v1_destroy(keyboard);
+
+  delete modifierState;
 }
 
-void WlrVirtualKeyboard::keyEvent(uint32_t keysym, uint32_t keycode, bool down)
+void WlrVirtualKeyboard::keyEvent(uint32_t /* keysym */, uint32_t keycode,
+                                  bool down)
 {
-  zwp_virtual_keyboard_v1_key(keyboard, keysym, keycode, down);
+  timespec ts;
+  uint32_t time;
+  bool updated;
+  if (keycode < code_map_qnum_to_xorgevdev_len)
+    keycode = code_map_qnum_to_xorgevdev[keycode];
+
+  // FIXME: look at keysym
```

Comment:
I guess this is a major difference between portals and wlroots? Should probably mention that somewhere.